src/shims/foreign_items/posix.rs

```diff
@@ -279,6 +266,77 @@ pub trait EvalContextExt<'mir, 'tcx: 'mir>: crate::MiriEvalContextExt<'mir, 'tcx
             => {
                 this.write_null(dest)?;
             }
+
+            "pthread_mutexattr_init" => {
+                let result = this.pthread_mutexattr_init(args[0])?;
+                this.write_scalar(Scalar::from_int(result, dest.layout.size), dest)?;
```

Comment:
I know most of this file uses `from_int` here, but that is a leftover from before we had the more specific helper methods. It seems the return type is `int` for all of them, so they could all become `Scalar::from_i32`?